python/mxnet/gluon/data/text/lm.py

```diff
@@ -0,0 +1,157 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=
+"""Language model datasets."""
+
+__all__ = ['LanguageModelDataset', 'WikiText2', 'WikiText103']
+
+import os
+import zipfile
+import shutil
+
+from .base import TextDataset, TextTokenReader
+from . import _constants as C
+from ...utils import download, check_sha1, get_repo_file_url
+
+
+class LanguageModelDataset(TextDataset):
```

Comment:
There is nothing special about this. I don't think we need a LanguageModelDataset class

Reply:
It's common for language modeling to preprocess arbitrary corpus into sequences of fixed number of tokens (which I call `collate` here). I put this preprocessing step in LangaugeModelDataset as it's not common for other datasets.

An alternative is to treat such preprocessing function as util function. Would that be preferred?